include/swift/SIL/TypeLowering.h

```diff
@@ -182,7 +190,9 @@ class TypeLowering {
       TypeExpansionSensitiveFlag = 1 << 4,
       InfiniteFlag               = 1 << 5,
       HasRawPointerFlag          = 1 << 6,
+      MoveOnlyFlag               = 1 << 7,
```

Comment:
I don't think we need to track "move-only" as a SIL-level recursive type property. The AST will need to know whether a type is move-only or not because of structural constraints already, in order to check generic constraints, so we should be able to ask the formal type directly whether it is move-only during type lowering.